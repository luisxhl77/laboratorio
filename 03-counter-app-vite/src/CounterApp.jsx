import { useState } from 'react';
import PropTypes from 'prop-types';


export const CounterApp = ({value}) => {
    //const [counter, setCounter] = useState(10);
    const [counter, setCounter] = useState(value);
    const contador =() => {
        setCounter( counter +1);
    }
    const restador = () => {
        setCounter( counter -1);
    }
    const reset = () => {
        setCounter(value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
        
            {/*<button onClick={contador}> +1 </button>*/}
            <button onClick={contador}> +1 </button>
            <button onClick={restador}> -1 </button>
            <button onClick={reset}> reset </button>
        </>
    );
}

// definicion del contenido de las propiedades
CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
}

// estados por default de las propiedades
CounterApp.defaultProps ={
    value: 123,
    name: 'luis',
}
