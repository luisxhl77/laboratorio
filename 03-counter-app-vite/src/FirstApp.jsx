
const titulo = "VIAJE AL CENTRO DE LA TIERRA";
const numeros = [1,2,3,4,5,6,7,8,9]
const objeto = {
  nombre: "luis",
  edad: 23,
}

const {nombre, edad} = objeto;
const saludo = () => {
  return ("hola a todos");
}

export const FirstApp = () => {
  return (
    <>
      <h1>{titulo}</h1>
      <h2>{numeros}</h2>
      <h3>{JSON.stringify( objeto )}</h3>
      <h4>{nombre + " " + edad}</h4>
      <h5>{saludo()}</h5>
    </>
  )
}

