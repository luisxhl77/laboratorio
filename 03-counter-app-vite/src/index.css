*{
    font-family: monospace;
    color: white;
    
}
body{
    background-color: black;
}
button{
    color: black;
    padding: 15px;
    font-size: 1.3rem;
    margin-right: 5px;
}