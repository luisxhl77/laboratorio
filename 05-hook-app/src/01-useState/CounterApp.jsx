import { useState } from "react"

export const CounterApp = () => {
    const [counter, setCounter] = useState({
      counter1: 10,
      counter2: 20,
      counter3: 30,
    });
    const {counter1, counter2, counter3,} = counter;
  return (
    <>
      <h1>COUNTER1:{counter1}</h1>
      <h1>COUNTER2:{counter2}</h1>
      <h1>COUNTER3:{counter3}</h1>
      <hr />
      
      <button onClick={ () => setCounter({
        ...counter,
        counter1: counter1 + 1,
      })
      }>+1</button>
      <button onClick={ () => setCounter({
        ...counter,
        counter3: counter3 - 1
      })}>-1</button>
    </>
  )
}
