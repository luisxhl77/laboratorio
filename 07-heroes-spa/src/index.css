*{
  margin: 0px;
  padding: 0px;
}
a{
  text-decoration: none;
  color: #b9b8b8;
}