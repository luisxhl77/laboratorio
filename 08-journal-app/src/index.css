*{
  margin: 0px;
  padding: 0px;
}
.box-shadow{
  box-shadow: 0px 5px 5px rgba(0, 0, 0, 0.2);
}