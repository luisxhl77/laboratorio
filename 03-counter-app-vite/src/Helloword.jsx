import PropTypes from 'prop-types';
import { FirstApp } from "./FirstApp";

export const Helloword = ({title, name}) => {
  return (
    <>
        <h1>{title}</h1>
        <h2>hola soy {name}</h2>
    </>
  )
}

Helloword.propTypes = {
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
}
Helloword.defaultProps = {
  title: 'none',
  name: 'none'
}