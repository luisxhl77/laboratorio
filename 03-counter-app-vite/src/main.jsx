import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { Helloword } from './Helloword';
import './index.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/*<Helloword title="hello word" name="goku"/>*/}
        <CounterApp value={1} name="luis"/>
    </React.StrictMode>
)


