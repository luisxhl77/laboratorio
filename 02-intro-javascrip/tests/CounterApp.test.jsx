import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas en el <CounterApp/>', ()=>{
    const initialValue = 100

    test('debe de hacer match con el snapshot',() => {

        const { container } = render( <CounterApp/>)
        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar el valor inicial de 100',() => {
         
        render( <CounterApp value= {initialValue}/>)
        expect( screen.getByText(initialValue) ).toBeTruthy();

    });

    test('debe de mostrar el valor inicial de 100 en el h2',() => {
         
        render( <CounterApp value= {initialValue}/>);

        expect( screen.getByRole('heading',{ level: 2}).innerHTML).toContain('100');

    });

    test('debe de incrementar al dar click en el boton +1',() => {
         
        render( <CounterApp value= {initialValue}/>);

        fireEvent.click( screen.getByLabelText('btnIncrement') )
        expect(screen.getByText('101')).toBeTruthy();

    });

    test('debe de decrementar al dar click en el boton -1',() => {
         
        render( <CounterApp value= {initialValue}/>);

        fireEvent.click( screen.getByLabelText('btnDecrement') )
        expect(screen.getByText('99')).toBeTruthy();

    });

    test('debe de resetear al dar click en el boton Reset',() => {
         
        render( <CounterApp value= {initialValue}/>);

        fireEvent.click( screen.getByLabelText('btnReset') )
        expect(screen.getByText('100')).toBeTruthy();

    });
});