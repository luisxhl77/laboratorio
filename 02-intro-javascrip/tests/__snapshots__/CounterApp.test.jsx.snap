// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
     
  </h2>
  <button
    aria-label="btnIncrement"
  >
     +1 
  </button>
  <button
    aria-label="btnDecrement"
  >
     -1 
  </button>
  <button
    aria-label="btnReset"
  >
     Reset 
  </button>
</div>
`;
